services/api/src/resources/project/harborSetup.js

```diff
@@ -151,6 +151,57 @@ const createHarborOperations = (sqlClient /* : MariaSQL */) => ({
     } catch (err) {
       logger.error(`Error while creating a webhook in the Harbor project for ${lagoonProjectName}, error: ${err}`)
     }
+  },
+  deleteProject: async (lagoonProjectName) => {
+    // Delete harbor project
+
+    // Get new harbor project's id
+    try {
+      const res = await harborClient.get(`projects?name=${lagoonProjectName}`)
+      var harborProjectID = res.body[0].project_id
+      logger.debug(`Got the harbor project id for project ${lagoonProjectName} successfully!`)
+    } catch (err) {
+      if (err.statusCode == 404) {
+        logger.warn(`Unable to get the harbor project id of "${lagoonProjectName}", as it does not exist in harbor!`)
+        return
+      } else {
+        logger.error(`Unable to get the harbor project id of "${lagoonProjectName}", error: ${err}`)
+      }
```

Comment:
you are catching an error here, but the code will continue below, is that intentionally? is the code below able to run without knowing the harborProjectID?

Reply:
Good catch. Trying to continue past not being able to find this value will lead to the API crashing.